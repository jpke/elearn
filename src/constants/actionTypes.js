export const LOADING = 'LOADING';
export const BAD_RESPONSE = 'BAD_RESPONSE';
export const LOG_IN = 'LOG_IN';
export const LOG_OUT = 'LOG_OUT';

export const SELECT_COURSE = 'SELECT_COURSE';
export const EDIT_COURSE = 'EDIT_COURSE';
export const ADD_USER = 'ADD_USER';
export const UPDATE_ENROLLABLE = 'UPDATE_ENROLLABLE';
export const UPDATE_ENROLLED = 'UPDATE_ENROLLED';
export const UPDATE_COURSE = 'UPDATE_COURSE';

export const LOAD_QUIZ = 'LOAD_QUIZ';
export const TOGGLE_QUIZ_VIEW = 'TOGGLE_QUIZ_VIEW';

export const CREATE_QUIZ = 'CREATE_QUIZ';
export const UPDATE_QUIZ = 'UPDATE_QUIZ';
export const DELETE_QUIZ = 'DELETE_QUIZ';
export const SAVE_QUIZ = 'SAVE_QUIZ';

export const START_QUIZ = 'START_QUIZ';
export const SELECT_ANSWER = 'SELECT_ANSWER';
export const NEXT_QUESTION = 'NEXT_QUESTION';
export const PREVIOUS_QUESTION = 'PREVIOUS_QUESTION';
export const SUBMIT_QUIZ = 'SUBMIT_QUIZ';
export const GET_LESSONS = 'GET_LESSONS';
export const GET_PDF = 'GET_PDF';
