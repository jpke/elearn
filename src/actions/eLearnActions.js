// import cookie from 'react-cookie'
import * as types from '../constants/actionTypes';

export function loading(item) {
  return {
    type: 'LOADING',
    item
  };
}

function loggedIn(response) {
  console.log("response: ", response);
  localStorage.setItem("login", {
    userName: response.name,
    user_Id: response._id,
    courses: response.courses,
    token: response.token
  });
  return {
    type: types.LOG_IN,
    userName: response.name,
    user_Id: response._id,
    courses: response.courses,
    token: response.token
  };
}

export function register(userName, email, password) {
  return function (dispatch) {
    dispatch(loading('register'));
    try {
      fetch('http://localhost:8080/elearn/users', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          'Accept': 'application/json'
        },
        body: JSON.stringify({
          name: userName,
          email: email,
          password: password
        })
      })
      .then(response => {
        if(response.status < 200 || response.status >= 300) {
          if(response.body.message === 'email already associated with an account') {
            return dispatch({
              type: types.EMAIL_USED,
              email
            });
          } else {
            let error = response;
            throw error;
          }
        }
        return response.json()
      })
      .then(response => {
        window.localStorage.loggedIn = true;
        dispatch(loading(''));
        dispatch(loggedIn(response));
      })
    } catch(error) {
      console.log("error response: ", error);
    }
  };
}

export function logIn(email, password) {
  return function (dispatch) {
    dispatch(loading('logIn'));
    try {
      fetch('http://localhost:8080/elearn/login', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          'Accept': 'application/json'
        },
        withCredentials: true,
        body: JSON.stringify({
          email: email,
          password: password
        })
      })
      .then(response => {
        if(response.status != 302) {
          let error = response;
          throw error;
        }
        return response.json()
      })
      .then(response => {
        window.localStorage.loggedIn = true;
        // console.log("set-cookie: ", response)
        // cookie.save("tokens",
        //   response.token,
        //   {
        //     httpOnly: true
        //   });
        dispatch(loading(''));
        dispatch(loggedIn(response));
      })
    } catch(error) {
      console.log("error response: ", error);
    }
  };
}

export function selectCourse(courseName, course_Id) {
  return {
    type: types.SELECT_COURSE,
    courseName,
    course_Id
  };
}

export function logOut() {
  //consider blacklisting token serverside in future
  window.localStorage.loggedIn = false;
  // cookie.remove('token');
  return {
    type: types.LOG_OUT
  };
}

export function startQuiz(token, course_Id) {
  return function (dispatch) {
    dispatch(loading('startQuiz'));
    //pull down quiz questions, then
    try {
      fetch('http://localhost:8080/elearn/quiz/'.concat(course_Id), {
        method: 'GET',
        headers: {
          'Content-Type': 'application/json',
          'Authorization': `Bearer ${token}`
        }
      })
      .then(response => {
        if(response.status < 200 || response.status >= 300) {
          let error = response;
          throw error;
        }
        return response.json()
      })
      .then(response => {
        dispatch(loading(''));
        console.log("quiz actions: ", response);
        return dispatch({
          type: types.START_QUIZ,
          quizData: response
        });
        })
    } catch(error) {
      console.log("error response: ", error);
    }
  };
}

export function selectAnswer(answerSelected, idSelected, item) {
  return {
    type: types.SELECT_ANSWER,
    answerSelected,
    idSelected,
    itemSelected: parseInt(item)
  };
}

export function nextQuestion() {
  return {
    type: types.NEXT_QUESTION
  };
}

export function prevQuestion() {
  return {
    type: types.PREVIOUS_QUESTION
  };
}

export function submitQuiz(quizData, quizTitle, quizId, _id, token) {
  return function (dispatch) {
    dispatch(loading('submitQuiz'));
    try {
      fetch('http://localhost:8080/elearn/quiz/submit', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          'Authorization': `Bearer ${token}`
        },
        body: JSON.stringify({
          title: quizTitle,
          quizData: quizData,
          quiz_Id: quizId,
          user_Id: _id
        })
      })
      .then(response => {
        if(response.status < 200 || response.status >= 300) {
          let error = response;
          throw error;
        } else {
          return response;
        }
      })
      .then(response => response.json())
      .then((response) => {
        console.log("action response: ", response);
        dispatch(loading(''));
        return dispatch({
          type: types.SUBMIT_QUIZ,
          score: response.score
        });
        })
    } catch(error) {
      console.log("error response: ", error);
    }
  }
}

export function getLessons(token) {
  return function (dispatch) {
    // let cookieToken = cookie.load('token');
    // console.log("cookieToken: ", cookieToken);
    dispatch(loading('lessons'));
    try {
      fetch('http://localhost:8080/elearn/lessons', {
        method: 'GET',
        headers: {
          'Content-Type': 'application/json',
          'Authorization': `Bearer ${token}`
        },
        // withCredentials: true,
      })
      .then(response => {
        if(response.status < 200 || response.status >= 300) {
          let error = response;
          throw error;
        }
        return response.json()
      })
      .then(response => {
        dispatch(loading(''));
        return dispatch({
          type: types.GET_LESSONS,
          lessons: response.entries
        });
        })
    } catch(error) {
      console.log("error response: ", error);
    }
  };
}

export function getPDF(pdfId, token) {
  return function (dispatch) {
    dispatch(loading("pdf"));
    try {
      fetch('http://localhost:8080/elearn/lessons/'.concat(pdfId), {
        method: 'GET',
        headers: {
          'Content-Type': 'application/json',
          'Authorization': `Bearer ${token}`
        },
        // withCredentials: true,
      })
      .then(response => {
        if(response.status < 200 || response.status >= 300) {
          let error = response;
          throw error;
        }
        return response.json()
      })
      .then(response => {
        dispatch(loading(""));
        console.log("response: ", response);
          return dispatch({
            type: types.GET_PDF,
            response
          });
        })
    } catch(error) {
      console.log("error response: ", error);
    }
  };
}
