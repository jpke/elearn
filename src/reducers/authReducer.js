import * as types from '../constants/actionTypes';

let initialState;
if(window.localStorage.userName) {
    initialState = {
      view: 'login',
      userName: window.localStorage.userName,
      userId: window.localStorage.user_Id,
      course: "",
      courses: JSON.parse(window.localStorage.courses),
      passed: window.localStorage.passed,
      token: window.localStorage.token,
      loggedIn: false,
      loading: false
    }
  } else {
    initialState = {
      view: 'login',
      userName: '',
      userId: '',
      course: "",
      courses: [],
      passed: [],
      token: "",
      loggedIn: false,
      loading: false
    }
  };

export default function authReducer(state = initialState, action) {
  let courses, index, i, passed;
  switch(action.type) {
    case types.LOADING:
      return {
        ...state,
        loading: !state.loading,
        loadingItem: action.item
      };
    case types.BAD_RESPONSE:
      return {
        errorMessage: action.message
      };
    case types.LOG_IN:
      return {
        ...state,
        userName: action.userName,
        userId: action.user_Id,
        courses: action.courses,
        course: "",
        passed: action.passed,
        token: action.token,
        loggedIn: true,
        url: action.url
      };
    case types.EMAIL_USED:
      return {
        ...state,
        emailUsed: true,
        email: action.email
      };
    case types.LOG_OUT:
      return {
        ...state,
        userName: "",
        userId: "",
        token: "",
        loggedIn: false
      };
    case types.SELECT_COURSE:
      return {
        ...state,
        course: action.course
      };
    case types.SUBMIT_QUIZ:
      if(action.passed) {
        passed = JSON.parse(JSON.stringify(state.passed)).concat({_id: action.attempt._id, of: action.attempt.of});
      } else {
        passed = state.passed;
      }
      return {
        ...state,
        passed: passed
      };
    case types.DELETE_QUIZ:
      index = -1;
      i = 0;
      for(i; i < action.courses.length; i++) {
        if(action.courses[i]._id === action.courseID) index = i;
      }
      return {
        ...state,
        course: action.courses[index],
        courses: action.courses
      };
    case types.SAVE_QUIZ:
      console.log("course: ", action.course);
      console.log("courses: ", state.courses);
      if(!action.course) return state;
      courses = JSON.parse(JSON.stringify(state.courses));
      index = -1;
      i = 0
      for(i; i < courses.length; i++) {
        if(courses[i]._id === action.course._id) index = i;
      }
      courses = courses.slice(0, index).concat(action.course, courses.slice(index + 1, courses.length));
      console.log("updated courses: ", courses);
      return {
        ...state,
        courses: courses,
        course: courses[index]
      };
    default:
    return state;
  }
}
