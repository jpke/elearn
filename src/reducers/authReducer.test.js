/* eslint-disable no-undef */
/* eslint-disable import/no-named-as-default */

import React from 'react';
import authReducer from './authReducer';
import {initialState} from './authReducer';
import * as types from '../constants/actionTypes';
import * as actions from '../actions/eLearnActions';

let state;
beforeEach(() => {
  state = JSON.parse(JSON.stringify(initialState));
})

it('should handle LOADING', () => {
  expect(authReducer(state, actions.loading(
    "sample item being loaded"
  ))).toMatchSnapshot();
})

it('should handle BAD_RESPONSE', () => {
  expect(authReducer(state, actions.badResponse(
    "sample error message"
  ))).toMatchSnapshot();
})

it('should handle LOG_IN', () => {
  expect(authReducer(state, actions.loggedIn(
    {
      name: "sample user",
      _id: "0000",
      courses: [
        {
          _id: "0000",
          name: "sample course",
          enrollable: "someUser@email.com",
          quizzes: [
            {
              _id: "1111",
              title: "sample quiz one"
            }
          ]
        }
      ],
      lessonFolder: [],
      admin: false
    }
  ))).toMatchSnapshot();
})

it('should handle LOG_OUT', () => {
  expect(authReducer(state, actions.logOut()
  )).toMatchSnapshot();
})

it('should handle SELECT_COURSE', () => {
  expect(authReducer(state, actions.selectCourse({
    _id: "0000",
    name: "sample course",
    enrollable: "someUser@email.com",
    quizzes: [
      {
        _id: "1111",
        title: "sample quiz one"
      }
    ]
  })
  )).toMatchSnapshot();
})

it('should handle EDIT_COURSE', () => {
  state.course = {
    _id: "0000",
    name: "sample course",
    enrollable: "someUser@email.com",
    quizzes: [
      {
        _id: "1111",
        title: "sample quiz one"
      }
    ]
  };
  expect(authReducer(state, actions.editCourse("course-title", "updated sample course title")
  )).toMatchSnapshot();
})

it('should handle UPDATE_ENROLLABLE', () => {
  expect(authReducer(state, {
    type: types.UPDATE_ENROLLABLE,
    enrollable: ["sampleUser@email.com"]
  }
  )).toMatchSnapshot();
})

it('should handle UPDATE_ENROLLED', () => {
  expect(authReducer(state, {
    type: types.UPDATE_ENROLLED,
    enrolled: ["sampleUser@email.com"]
  }
  )).toMatchSnapshot();
})

it('should handle UPDATE_COURSE', () => {
  state.courses = [
    {
      _id: "0000",
      name: "sample course",
      enrollable: "someUser@email.com",
      quizzes: [
        {
          _id: "1111",
          title: "sample quiz one"
        }
      ]
    },
    {
      _id: "0001",
      name: "sample course two",
      enrollable: "someUser@email.com",
      quizzes: [
        {
          _id: "1112",
          title: "sample quiz one"
        }
      ]
    }
  ];
  expect(authReducer(state, {
    type: types.UPDATE_COURSE,
    course: {
      _id: "0000",
      name: "updated sample course",
      enrollable: "someUser@email.com",
      quizzes: [
        {
          _id: "1111",
          title: "sample quiz one"
        }
      ]
    }
  }
  )).toMatchSnapshot();
})

it('should handle passing SUBMIT_QUIZ', () => {
  expect(authReducer(state, {
    type: types.SUBMIT_QUIZ,
    score: 3,
    attempt: {
      of: "0000",
      score: 3,
      user: "0000",
      _id: "1111",
      item: [
        {
          question: "question one",
          _id: "aaaa",
          correct: true,
          idSelected: "1a1a",
          itemSelected: 1,
          answers: [
            {
              correct: true,
              answer: "correct answer",
              _id: "1a1a"
            },
            {
              correct: false,
              answer: "correct answer",
              _id: "1a1b"
            },
          ]
        },
        {
          question: "question two",
          _id: "aaab",
          correct: true,
          idSelected: "1a1a",
          itemSelected: 1,
          answers: [
            {
              correct: true,
              answer: "correct answer",
              _id: "1a1a"
            },
            {
              correct: false,
              answer: "correct answer",
              _id: "1a1b"
            },
          ]
        },
        {
          question: "question three",
          _id: "aaac",
          correct: true,
          idSelected: "1a1a",
          itemSelected: 1,
          answers: [
            {
              correct: true,
              answer: "correct answer",
              _id: "1a1a"
            },
            {
              correct: false,
              answer: "correct answer",
              _id: "1a1b"
            },
          ]
        }
      ]
    },
    passed: {
      _id: "aaaa",
      of: "0000"
    }
  }
  )).toMatchSnapshot();
})

it('should handle non-passing SUBMIT_QUIZ', () => {
  expect(authReducer(state, {
    type: types.SUBMIT_QUIZ,
    score: 1,
    attempt: {
      of: "0000",
      score: 1,
      user: "0000",
      _id: "1111",
      item: [
        {
          question: "question one",
          _id: "aaaa",
          correct: true,
          idSelected: "1a1a",
          itemSelected: 1,
          answers: [
            {
              correct: true,
              answer: "correct answer",
              _id: "1a1a"
            },
            {
              correct: false,
              answer: "correct answer",
              _id: "1a1b"
            },
          ]
        },
        {
          question: "question two",
          _id: "aaab",
          correct: false,
          idSelected: "1a1a",
          itemSelected: 1,
          answers: [
            {
              correct: true,
              answer: "correct answer",
              _id: "1a1a"
            },
            {
              correct: false,
              answer: "correct answer",
              _id: "1a1b"
            },
          ]
        },
        {
          question: "question three",
          _id: "aaac",
          correct: false,
          idSelected: "1a1a",
          itemSelected: 1,
          answers: [
            {
              correct: true,
              answer: "correct answer",
              _id: "1a1a"
            },
            {
              correct: false,
              answer: "correct answer",
              _id: "1a1b"
            },
          ]
        }
      ]
    },
    passed: false
  }
  )).toMatchSnapshot();
})

it('should handle DELETE_QUIZ', () => {
  expect(authReducer(state, {
    type: types.DELETE_QUIZ,
    courses: [
      {
        _id: "0000",
        name: "sample course",
        enrollable: "someUser@email.com",
        quizzes: [
          {
            _id: "1111",
            title: "sample quiz one"
          }
        ]
      },
      {
        _id: "0001",
        name: "sample course two",
        enrollable: "someUser@email.com",
        quizzes: [
          {
            _id: "1112",
            title: "sample quiz one"
          }
        ]
      }
    ],
    courseID: "0000"
  }
  )).toMatchSnapshot();
})

it('should handle SAVE_QUIZ', () => {
  state.courses = [
    {
      _id: "0000",
      name: "sample course",
      enrollable: "someUser@email.com",
      quizzes: [
        {
          _id: "1111",
          title: "sample quiz one"
        }
      ]
    },
    {
      _id: "0001",
      name: "sample course two",
      enrollable: "someUser@email.com",
      quizzes: [
        {
          _id: "1112",
          title: "sample quiz one"
        }
      ]
    }
  ];
  expect(authReducer(state, {
    type: types.SAVE_QUIZ,
    course: {
        _id: "0000",
        name: "sample course",
        enrollable: "someUser@email.com",
        quizzes: [
          {
            _id: "1111",
            title: "sample quiz one updated"
          }
        ]
    }
  }
  )).toMatchSnapshot();
})
