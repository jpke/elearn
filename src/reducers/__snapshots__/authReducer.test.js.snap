exports[`test should handle BAD_RESPONSE 1`] = `
Object {
  "course": "",
  "courses": Array [],
  "enrollable": Array [],
  "enrolled": Array [],
  "errorMessage": "sample error message",
  "loading": false,
  "loggedIn": false,
  "message": "",
  "newUser": "",
  "passed": Array [],
  "token": "",
  "userId": "",
  "userName": "",
  "view": "login",
}
`;

exports[`test should handle DELETE_QUIZ 1`] = `
Object {
  "course": Object {
    "_id": "0000",
    "enrollable": "someUser@email.com",
    "name": "sample course",
    "quizzes": Array [
      Object {
        "_id": "1111",
        "title": "sample quiz one",
      },
    ],
  },
  "courses": Array [
    Object {
      "_id": "0000",
      "enrollable": "someUser@email.com",
      "name": "sample course",
      "quizzes": Array [
        Object {
          "_id": "1111",
          "title": "sample quiz one",
        },
      ],
    },
    Object {
      "_id": "0001",
      "enrollable": "someUser@email.com",
      "name": "sample course two",
      "quizzes": Array [
        Object {
          "_id": "1112",
          "title": "sample quiz one",
        },
      ],
    },
  ],
  "enrollable": Array [],
  "enrolled": Array [],
  "loading": false,
  "loggedIn": false,
  "message": "",
  "newUser": "",
  "passed": Array [],
  "token": "",
  "userId": "",
  "userName": "",
  "view": "login",
}
`;

exports[`test should handle EDIT_COURSE 1`] = `
Object {
  "course": Object {
    "_id": "0000",
    "enrollable": "someUser@email.com",
    "name": "updated sample course title",
    "quizzes": Array [
      Object {
        "_id": "1111",
        "title": "sample quiz one",
      },
    ],
  },
  "courses": Array [],
  "enrollable": Array [],
  "enrolled": Array [],
  "loading": false,
  "loggedIn": false,
  "message": "",
  "newUser": "",
  "passed": Array [],
  "token": "",
  "userId": "",
  "userName": "",
  "view": "login",
}
`;

exports[`test should handle LOADING 1`] = `
Object {
  "course": "",
  "courses": Array [],
  "enrollable": Array [],
  "enrolled": Array [],
  "loading": true,
  "loadingItem": "sample item being loaded",
  "loggedIn": false,
  "message": "",
  "newUser": "",
  "passed": Array [],
  "token": "",
  "userId": "",
  "userName": "",
  "view": "login",
}
`;

exports[`test should handle LOG_IN 1`] = `
Object {
  "course": "",
  "courses": Array [
    Object {
      "_id": "0000",
      "enrollable": "someUser@email.com",
      "name": "sample course",
      "quizzes": Array [
        Object {
          "_id": "1111",
          "title": "sample quiz one",
        },
      ],
    },
  ],
  "enrollable": Array [],
  "enrolled": Array [],
  "loading": false,
  "loggedIn": true,
  "message": "",
  "newUser": "",
  "passed": undefined,
  "token": "sampleRandomToken",
  "url": "http://erudome.us-east-1.elasticbeanstalk.com/elearn/",
  "userId": "0000",
  "userName": "sample user",
  "view": "login",
}
`;

exports[`test should handle LOG_OUT 1`] = `
Object {
  "course": "",
  "courses": Array [],
  "enrollable": Array [],
  "enrolled": Array [],
  "loading": false,
  "loggedIn": false,
  "message": "",
  "newUser": "",
  "passed": Array [],
  "token": "",
  "userId": "",
  "userName": "",
  "view": "login",
}
`;

exports[`test should handle SAVE_QUIZ 1`] = `
Object {
  "course": Object {
    "_id": "0000",
    "enrollable": "someUser@email.com",
    "name": "sample course",
    "quizzes": Array [
      Object {
        "_id": "1111",
        "title": "sample quiz one updated",
      },
    ],
  },
  "courses": Array [
    Object {
      "_id": "0000",
      "enrollable": "someUser@email.com",
      "name": "sample course",
      "quizzes": Array [
        Object {
          "_id": "1111",
          "title": "sample quiz one updated",
        },
      ],
    },
    Object {
      "_id": "0001",
      "enrollable": "someUser@email.com",
      "name": "sample course two",
      "quizzes": Array [
        Object {
          "_id": "1112",
          "title": "sample quiz one",
        },
      ],
    },
  ],
  "enrollable": Array [],
  "enrolled": Array [],
  "loading": false,
  "loggedIn": false,
  "message": "",
  "newUser": "",
  "passed": Array [],
  "token": "",
  "userId": "",
  "userName": "",
  "view": "login",
}
`;

exports[`test should handle SELECT_COURSE 1`] = `
Object {
  "course": Object {
    "_id": "0000",
    "enrollable": "someUser@email.com",
    "name": "sample course",
    "quizzes": Array [
      Object {
        "_id": "1111",
        "title": "sample quiz one",
      },
    ],
  },
  "courses": Array [],
  "enrollable": Array [],
  "enrolled": Array [],
  "loading": false,
  "loggedIn": false,
  "message": "",
  "newUser": "",
  "passed": Array [],
  "token": "",
  "userId": "",
  "userName": "",
  "view": "login",
}
`;

exports[`test should handle UPDATE_COURSE 1`] = `
Object {
  "course": Object {
    "_id": "0000",
    "enrollable": "someUser@email.com",
    "name": "updated sample course",
    "quizzes": Array [
      Object {
        "_id": "1111",
        "title": "sample quiz one",
      },
    ],
  },
  "courses": Array [
    Object {
      "_id": "0000",
      "enrollable": "someUser@email.com",
      "name": "updated sample course",
      "quizzes": Array [
        Object {
          "_id": "1111",
          "title": "sample quiz one",
        },
      ],
    },
    Object {
      "_id": "0001",
      "enrollable": "someUser@email.com",
      "name": "sample course two",
      "quizzes": Array [
        Object {
          "_id": "1112",
          "title": "sample quiz one",
        },
      ],
    },
  ],
  "enrollable": Array [],
  "enrolled": Array [],
  "loading": false,
  "loggedIn": false,
  "message": "Course saved!",
  "newUser": "",
  "passed": Array [],
  "token": "",
  "userId": "",
  "userName": "",
  "view": "login",
}
`;

exports[`test should handle UPDATE_ENROLLABLE 1`] = `
Object {
  "course": "",
  "courses": Array [],
  "enrollable": Array [
    "sampleUser@email.com",
  ],
  "enrolled": Array [],
  "loading": false,
  "loggedIn": false,
  "message": "",
  "newUser": "",
  "passed": Array [],
  "token": "",
  "userId": "",
  "userName": "",
  "view": "login",
}
`;

exports[`test should handle UPDATE_ENROLLED 1`] = `
Object {
  "course": "",
  "courses": Array [],
  "enrollable": Array [],
  "enrolled": Array [
    "sampleUser@email.com",
  ],
  "loading": false,
  "loggedIn": false,
  "message": "",
  "newUser": "",
  "passed": Array [],
  "token": "",
  "userId": "",
  "userName": "",
  "view": "login",
}
`;

exports[`test should handle non-passing SUBMIT_QUIZ 1`] = `
Object {
  "course": "",
  "courses": Array [],
  "enrollable": Array [],
  "enrolled": Array [],
  "loading": false,
  "loggedIn": false,
  "message": "",
  "newUser": "",
  "passed": Array [],
  "token": "",
  "userId": "",
  "userName": "",
  "view": "login",
}
`;

exports[`test should handle passing SUBMIT_QUIZ 1`] = `
Object {
  "course": "",
  "courses": Array [],
  "enrollable": Array [],
  "enrolled": Array [],
  "loading": false,
  "loggedIn": false,
  "message": "",
  "newUser": "",
  "passed": Array [
    Object {
      "_id": "1111",
      "of": "0000",
    },
  ],
  "token": "",
  "userId": "",
  "userName": "",
  "view": "login",
}
`;
